// Copyright (c) 2021, The MathWorks, Inc.

syntax = "proto3";

package mathworks.scenario.scene.hdmap;

import "mathworks/scenario/common/geometry.proto";

// The following schema defines general information about a RoadRunner  
// HD map payload, including data bounds and projection information.
// A RoadRunner HD map asset file will be in the following format:
// (Varint) HeaderSize > Header Message > RoadRunner HD Map Message
// The Header Message should be delimited by the size of the message
// More information on encodings and Varint size can be referred here: 
//    https://developers.google.com/protocol-buffers/docs/encoding
//    https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.io.coded_stream
//
message Header
{
  string author = 1;
  scenario.common.Projection projection = 2;
  DataBounds geographic_boundary = 3;
}

//
// Spatial bounds of the geometric data stored in the transfer map, 
// represented in the coordinate space of the transfer mapâ€™s projection.
// Primarily intended as an optimization for spatial queries 
// like to determine whether a Transfer Map payload intersects a given spatial region
//
message DataBounds
{ 
  scenario.common.Box3 bounds = 1;
}
